### PR TITLE
get interface building and working with clang-7 on ubuntu 18

### PR DESCRIPTION
**Pull request by sethalves**
- get interface building and working with clang-7 on ubuntu 18

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- the only functional change is around how the ability to reposition windows on the desktop is enabled, so start interface (launching into desktop), open and close some windows, move them around
- also launch into HMD and check this

File: interface/src/raypick/PathPointer.h [hunk 1]
```diff
@@ -27,6 +27,7 @@ class StartEndRenderState {
 public:
     StartEndRenderState() {}
     StartEndRenderState(const OverlayID& startID, const OverlayID& endID);
+    virtual ~StartEndRenderState() {}
```
Comment (jherico) [inline interface/src/raypick/PathPointer.h]: not a big deal, but you can also declare these as `virtual ~StartEndRenderState() = default;`

File: libraries/ui/src/OffscreenUi.cpp [hunk 2]
```diff
@@ -205,9 +205,12 @@ bool OffscreenUi::isPointOnDesktopWindow(QVariant point) {
 }
 
 void OffscreenUi::hide(const QString& name) {
-    QQuickItem* item = getRootItem()->findChild<QQuickItem*>(name);
-    if (item) {
-        QQmlProperty(item, OFFSCREEN_VISIBILITY_PROPERTY).write(false);
+    auto rootItem = getRootItem();
+    if (rootItem) {
+        QQuickItem* item = getRootItem()->findChild<QQuickItem*>(name);
```
Comment (jherico) [inline libraries/ui/src/OffscreenUi.cpp]: this should re-use `rootItem`

File: tests-manual/gpu-textures/src/main.cpp [hunk 3]
```diff
@@ -85,14 +85,12 @@ class MyTestWindow : public TestWindow {
 
     void updateCamera() {
         float t = _time.elapsed() * 1e-3f;
-        glm::vec3 unitscale{ 1.0f };
         glm::vec3 up{ 0.0f, 1.0f, 0.0f };
```
Comment (jherico) [inline tests-manual/gpu-textures/src/main.cpp]: can be declared static const

File: tests-manual/gpu/src/main.cpp [hunk 4]
```diff
@@ -102,14 +102,12 @@ class MyTestWindow : public TestWindow {
 #ifdef INTERACTIVE
         t = _time.elapsed() * 1e-3f;
 #endif
-        glm::vec3 unitscale { 1.0f };
         glm::vec3 up { 0.0f, 1.0f, 0.0f };
```
Comment (jherico) [inline tests-manual/gpu/src/main.cpp]: can be declared static const

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13470/HighFidelity-Beta-PR14110-cb04894.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13470/HighFidelity-Beta-Interface-PR14110-cb04894.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13470/HighFidelity-Beta-PR14110-cb04894.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13470/HighFidelity-Beta-Interface-PR14110-cb04894.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13475/HighFidelity-Beta-PR14110-bbe9b5c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13475/HighFidelity-Beta-PR14110-bbe9b5c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13475/HighFidelity-Beta-Interface-PR14110-bbe9b5c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13475/HighFidelity-Beta-PR14110-bbe9b5c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13475/HighFidelity-Beta-Interface-PR14110-bbe9b5c.exe

**Comment by jakram** [comment]
Currently testing

**Comment by jakram** [comment]
@conklin94122 PASS

Tested on Desktop and HMD in Windows and Desktop in MacOS. The windows move and display correctly. Cannot get the windows to crash by switching through them, closing and opening the interface, or resizing the Interface window.

Additionally ran a smoke test to verify that the app ran correctly on both platforms.

**Pull request merged**